CenterForOpenScience/ember-osf-web
# [EMB-342] Developer apps

(based on #387 -- merge that first or review these together)

## Purpose
Add developer apps page


## Summary of Changes
- Add developer apps settings page
- Add `validated-model-form` component, use it for editing/creating developer apps and personal access tokens

Create:
![developer app -- create](https://user-images.githubusercontent.com/6776190/45714271-1eedfa80-bb5f-11e8-825f-c5937f19d339.gif)

Edit:
![developer app -- edit](https://user-images.githubusercontent.com/6776190/45714286-26ad9f00-bb5f-11e8-8e44-79c778c46a2d.gif)

Delete:
![developer app -- delete](https://user-images.githubusercontent.com/6776190/45714312-375e1500-bb5f-11e8-986a-246a0219be60.gif)

Reset client secret:
![developer app -- reset](https://user-images.githubusercontent.com/6776190/45714327-45ac3100-bb5f-11e8-838f-1c349992db30.gif)


## Side Effects
Changed some internals of the tokens page, should be no visible effect.


## Feature Flags
`ember_user_settings_apps_page`


## QA Notes
- Create, edit, delete, reset developer apps at `/settings/applications`
- Validation FYI:
  - Name/URLs: required, max 200 characters
  - Description: optional, max 1000 characters
  - URLs need to be valid URLs, but no other checking is done
- Check that resetting a client secret actually results in a new secret
- Worth a quick regression of the tokens create/edit forms, just in case


## Ticket


https://openscience.atlassian.net/browse/EMB-342

# Reviewer Checklist
